app/Model/TaskLink.php

```diff
@@ -0,0 +1,342 @@
+        $this->db->startTransaction();
+        $res = $this->db->table(self::TABLE)
+            ->eq('id', $values['id'])
+            ->save($taskLink1);
+        if (! $res) {
+            $this->db->cancelTransaction();
+            return false;
+        }
+        $res = $this->db->table(self::TABLE)
+            ->eq('id', (0 == ($values['id'] % 2)) ? $values['id'] - 1 : $values['id'] + 1)
+            ->save($taskLink2);
+        if (! $res) {
+            $this->db->cancelTransaction();
+            return false;
+        }
+        $this->db->closeTransaction();
+        return true;
+    }
+
+    /**
+     * Remove a link
+     *
+     * @access public
+     * @param integer $id
+     *            Link id
+     * @return bool
+     */
+    public function remove($id)
+    {
+        $this->db->startTransaction();
+        $res = $this->db->table(self::TABLE)
+            ->eq('id', $id)
+            ->remove();
+        if (! $res) {
+            $this->db->cancelTransaction();
+            return false;
+        }
+        $res = $this->db->table(self::TABLE)
+            ->eq('id', (0 == ($id % 2)) ? $id - 1 : $id + 1)
+            ->remove();
+        if (! $res) {
+            $this->db->cancelTransaction();
+            return false;
+        }
+        $this->db->closeTransaction();
+        return true;
+    }
+
+    /**
+     * Duplicate all links to another task
+     *
+     * @access public
+     * @param integer $src_task_id
+     *            Source task id
+     * @param integer $dst_task_id
+     *            Destination task id
+     * @return bool
+     */
+    public function duplicate($src_task_id, $dst_task_id)
+    {
+        return $this->db->transaction(function ($db) use($src_task_id, $dst_task_id)
+        {
+            $links = $db->table(TaskLink::TABLE)
+                ->columns('link_id', 'task_id', 'task_inverse_id')
+                ->eq('task_id', $src_task_id)
+                ->asc('id')
+                ->findAll();
+            foreach ($links as &$link) {
+                $link['task_id'] = $dst_task_id;
+                if (! $db->table(TaskLink::TABLE)
+                    ->save($link)) {
+                    return false;
+                }
+            }
+            
+            $links = $db->table(TaskLink::TABLE)
+                ->columns('link_id', 'task_id', 'task_inverse_id')
+                ->eq('task_inverse_id', $src_task_id)
+                ->asc('id')
+                ->findAll();
+            foreach ($links as &$link) {
+                $link['task_inverse_id'] = $dst_task_id;
+                if (! $db->table(TaskLink::TABLE)
+                    ->save($link)) {
+                    return false;
+                }
+            }
+        });
+    }
+
+    public function isUniqueConstraintFailed()
+    {
+        return $this->search($this->db->getLogMessages(), 'UNIQUE') || $this->search($this->db->getLogMessages(), 'Duplicata');
+    }
```

Comment:
You should not do that. That will work only in debug mode. Check if the link exists at the validation.